masonry/src/core/contexts.rs

```diff
@@ -780,6 +780,23 @@ impl_context_method!(
     }
 );
 
+// Methods on all context types except LayoutCtx and ComposeCtx.
+// Access selected global state information.
+impl_context_method!(
+    MutateCtx<'_>,
+    QueryCtx<'_>,
+    EventCtx<'_>,
+    UpdateCtx<'_>,
+    PaintCtx<'_>,
+    AccessCtx<'_>,
+    {
+        /// Get DPI scaling factor.
+        pub fn get_scale_factor(&self) -> f64 {
+            self.global_state.scale_factor
+        }
+    }
+);
+
```

Comment:
Given that changing the scale factor only causes a re-render, I would recommend only giving access to it in PaintCtx and AccessCtx. Basically assume it's an implementation detail of the rendering and it's not accessible to app logic.

Reply:
limited to 2